carbonmark/components/pages/Retirements/RetirementProvenance/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { ProvenanceRecord } from ".generated/carbonmark-api-sdk/types";
+import { GridContainer, Section } from "@klimadao/lib/components";
+import { concatAddress, formatTonnes } from "@klimadao/lib/utils";
+import { Trans, t } from "@lingui/macro";
+import { Footer } from "components/Footer";
+import { PageHead } from "components/PageHead";
+import { SocialLinks } from "components/SocialLinks";
+import { Text } from "components/Text";
+import { Navigation } from "components/shared/Navigation";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import { RetirementProvenancePageProps } from "pages/retirements/[beneficiary]/[retirement_index]/provenance";
+import { ProvenanceComponent } from "./ProvenanceComponent";
+import { RetirementCard } from "./RetirementCard";
+import * as styles from "./styles";
+
+/**
+ * Provenance Elements indexed by the address of the last sender
+ */
+type ProvenanceList = Record<string, ProvenanceRecord[]>;
+
+/**
+ * Splits Provenance records in different streams
+ * Each stream tracks the provenance of the credits received by the last sender
+ * @param provenance
+ * @returns
+ */
+const splitProvenance = (records: ProvenanceRecord[]) => {
```

Comment:
Again, would be great in a `.utils` file

Reply:
Done in 87b97bd1f